meta-genivi-dev/meta-genivi-dev/recipes-extended/minifi-cpp/minifi-cpp_0.5.0.bb

```diff
@@ -0,0 +1,81 @@
+# built and there is a lot of references to include directories in the various
+# cmake files to "thirdpart".
+#
+# The proper clean-up would be to actually create a meta-minifi which provides
+# the recipes for all the dependecies that are not avaiable in upstream
+# meta layers.
+
+
+SUMMARY = "Apache NiFi - MiNiFi C++ is a complementary data collection \
+approach that supplements the core tenets of NiFi in dataflow management, \
+focusing on the collection of data at the source of its creation. The C++ \
+implementation is an additional implementation to the one in Java with the \
+aim of an even smaller resource footprint."
+
+LICENSE = "Unknown & MIT & GPLv2 & Apache-2.0 & EDL-1.0"
+LIC_FILES_CHKSUM = "\
+    file://LICENSE;md5=b58267cabd65636363f906f126095ff7 \
+    file://thirdparty/rapidjson-1.1.0/license.txt;md5=ba04aa8f65de1396a7e59d1d746c2125 \
+    file://thirdparty/uuid/COPYING;md5=58dcd8452651fc8b07d1f65ce07ca8af \
+    file://thirdparty/yaml-cpp-yaml-cpp-20171024/LICENSE;md5=6a8aaf0595c2efc1a9c2e0913e9c1a2c \
+    file://thirdparty/date/LICENSE.txt;md5=b5d973344b3c7bbf7535f0e6e002d017 \
+    file://thirdparty/spdlog-20170710/LICENSE;md5=6e5242b8f24d08c5e948675102937cc9 \
+    file://thirdparty/cxxopts/LICENSE;md5=8de00431559a76a1b43f6fd44f8f6689 \
+    file://thirdparty/civetweb-1.10/LICENSE.md;md5=da079d81be91ff1d1ca56dcd751f897f \
+    file://thirdparty/pcap++/LICENSE;md5=911690f51af322440237a253d695d19f \
+    file://thirdparty/pcap++/3rdParty/dirent-for-Visual-Studio/LICENSE;md5=b1177b216c5b9b15e89b9612a05ca461 \
+    file://thirdparty/pcap++/3rdParty/LightPcapNg/LightPcapNg/LICENSE.txt;md5=3b79e2c09be0f47e2543c923ea2a0bd7 \
+    file://thirdparty/libuvc-0.0.6/LICENSE.txt;md5=2f1963e0bb88c93463af750daf9ba0c2 \
+"
+
+SRC_URI = "\
+    https://github.com/apache/nifi-minifi-cpp/archive/rel/minifi-cpp-${PV}.tar.gz \
+    file://0001-MINIFICPP-562-Fixed-thirdparty-uuid-include-path.patch \
+    file://0002-thirdparty-date-remove-usage-of-SYSTEM.patch \
+"
+
+SRC_URI[md5sum] = "2a63fd6692a55db2fedc99b986da32b4"
+SRC_URI[sha256sum] = "b7fb566706504c660ce43cd466ede58b649e94fb51f57dab8be02eabf6ea0f73"
+
+S = "${WORKDIR}/nifi-${BPN}-rel-${BPN}-${PV}"
+
+DEPENDS = "\
+    bison \
+    bzip2 \
+    curl \
+    flex \
+    icu \
+    openssl \
+    python3 \
+    zlib \
+"
+
+RDEPENDS_${PN} = "\
+    libcurl \
+"
+
+PACKAGECONFIG ?= "rocksdb"
+
+PACKAGECONFIG[mqtt] = "-DENABLE_MQTT=ON,-DENABLE_MQTT=OFF,paho-mqtt-c"
+PACKAGECONFIG[rocksdb] = "-DDISABLE_ROCKSDB=OFF,-DDISABLE_ROCKSDB=ON, rocksdb"
+PACKAGECONFIG[libarchive] = "-DDISABLE_LIBARCHIVE=OFF,-DDISABLE_LIBARCHIVE=ON,libarchive"
+PACKAGECONFIG[gps] = "-DENABLE_GPS=ON,-DENABLE_GPS=OFF,"
+
+RDEPENDS_${PN} += "${@bb.utils.contains('PACKAGECONFIG', 'gps', 'libgps', '', d)}"
+
+EXTRA_OECMAKE = "-DSKIP_TESTS=ON -DBUILD_ROCKSDB=OFF -DUSE_SYSTEM_UUID=OFF"
+
+inherit cmake pkgconfig
+
+do_install_append() {
+     rm ${D}/usr/NOTICE ${D}/usr/README.md ${D}/usr/LICENSE
```

Comment:
please add -rf option not to get the error in second build time.

Reply:
Nope, this is by design and should fail if one of the files does not exist. This will indicate that something has changed in the upstream package.

There is a dependency chain in place here, meaning that do_install_append will always run after do_install meaning that these files will always be present on the current revision. There will not be any problems with "second build" because if the package is re-built that means that do_install will run, or if the sstate-cache is re-used then it wont run any of the steps.